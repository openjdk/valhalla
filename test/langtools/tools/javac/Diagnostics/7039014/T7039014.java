/*
 * @test /nodynamiccopyright/
 * @bug     7039014
 * @summary Confusing error message for method conflict
 *
 * @compile/fail/ref=T7039014.out -XDrawDiagnostics T7039014.java
 */
public class T7039014 {

// Test 1: this should fail to compile

    interface A1<T> {
        byte m(String x);
        char m(T x);
    }

    interface B1 extends A1<String> {
    }

// Test 2: this should fail to compile

    interface A2<T> {
        default byte m(String x) { return 0; }
        char m(T x);
    }

    interface B2 extends A2<String> {
    }

// Test 3: this should compile

    interface A3<T> {
        private byte m(String x) { return 0; }
        char m(T x);
    }

    interface B3 extends A3<String> {
    }
}
