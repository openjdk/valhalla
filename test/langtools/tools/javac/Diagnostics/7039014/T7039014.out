T7039014.java:17:5: compiler.err.incompatible.diff.ret.same.type: T7039014.A1<java.lang.String>, m, java.lang.String
T7039014.java:27:5: compiler.err.incompatible.diff.ret.same.type: T7039014.A2<java.lang.String>, m, java.lang.String
2 errors
